### PR TITLE
[BACKLOG-41134] [BACKLOG-40475] Disable Folder Actions under Repository

### PR DESCRIPTION
**Pull request by singletonc**
- Disable all Folder Actions for root folders "Repository" and "VFS Connections"
- Hide icons for root folders, to communicate to users that they shouldn't be considered standard folders
- Fix title == null conditional. This should have originally been handled in browser.js reformatResponse and not done in the template.

### CODE REVIEW

File: user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js [hunk 1]
```diff
@@ -716,10 +722,14 @@ define([
...
 
+          if( isProviderRootPath(obj.file.path) ){
+            obj.file.isProviderRootPath = true;
+          }
+
           //decode potentially encoded name and title attributes of PVFS paths
           if( !isRepositoryPath(obj.file.path) && !obj.file.decoded ){
             obj.file.name = this.decodePvfsFileAttribute(obj.file.name);
-            obj.file.title = obj.file.title === null ? null : this.decodePvfsFileAttribute(obj.file.title);
+            obj.file.title = obj.file.title === null ? obj.file.name : this.decodePvfsFileAttribute(obj.file.title);
```
Comment (singletonc) [inline user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js:732]: I should have been doing this here before, not in the template.

File: user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.templates.js [hunk 2]
```diff
@@ -91,10 +91,10 @@ define([
           "{{#ifCond file.path '.trash'}}" +
           "<div id='{{file.objectId}}' class='trash folder' path='{{file.path}}' ext='{{file.name}}' desc='{{file.name}}'>" +
           "{{else}}" +
-              "{{#if file.title}}" +
```
Comment (singletonc) [inline user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.templates.js:94]: I should have been handling the title in browser.js reformatresponse and not in the template. I fixed it here as I decided to handle the element class via template conditional.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 13m 20s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5641/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 43s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5641/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[🚨 Frogbot scanned this pull request and found the below:](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>


## 📦 Vulnerable Dependencies
### ✍️ Summary
<div align='center'>

| SEVERITY                | DIRECT DEPENDENCIES                  | IMPACTED DEPENDENCY                  | FIXED VERSIONS                  | CVES                  |
| :---------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: |
| <br>  Medium | org.apache.poi:poi-scratchpad:4.1.1<br>pentaho:pentaho-platform-repository:10.2.0.0-SNAPSHOT<br>pentaho:pentaho-platform-extensions:10.2.0.0-SNAPSHOT | org.apache.poi:poi-scratchpad 4.1.1 | [5.2.1] | CVE-2022-26336 |

</div>

### 🔬 Research Details
**Description:**
A shortcoming in the HMEF package of poi-scratchpad (Apache POI) allows an attacker to cause an Out of Memory exception. This package is used to read TNEF files (Microsoft Outlook and Microsoft Exchange Server). If an application uses poi-scratchpad to parse TNEF files and the application allows untrusted users to supply them, then a carefully crafted file can cause an Out of Memory exception. This issue affects poi-scratchpad version 5.2.0 and prior versions. Users are recommended to upgrade to poi-scratchpad 5.2.1.
<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :x: Build failed in 9m 24s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5641/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[🚨 Frogbot scanned this pull request and found the below:](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>


## 📦 Vulnerable Dependencies
### ✍️ Summary
<div align='center'>

| SEVERITY                | DIRECT DEPENDENCIES                  | IMPACTED DEPENDENCY                  | FIXED VERSIONS                  | CVES                  |
| :---------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: |
| <br>  Medium | org.apache.poi:poi-scratchpad:4.1.1<br>pentaho:pentaho-platform-repository:10.2.0.0-SNAPSHOT<br>pentaho:pentaho-platform-extensions:10.2.0.0-SNAPSHOT | org.apache.poi:poi-scratchpad 4.1.1 | [5.2.1] | CVE-2022-26336 |

</div>

### 🔬 Research Details
**Description:**
A shortcoming in the HMEF package of poi-scratchpad (Apache POI) allows an attacker to cause an Out of Memory exception. This package is used to read TNEF files (Microsoft Outlook and Microsoft Exchange Server). If an application uses poi-scratchpad to parse TNEF files and the application allows untrusted users to supply them, then a carefully crafted file can cause an Out of Memory exception. This issue affects poi-scratchpad version 5.2.0 and prior versions. Users are recommended to upgrade to poi-scratchpad 5.2.1.
<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
<div align='center'>

[🚨 Frogbot scanned this pull request and found the below:](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>


## 📦 Vulnerable Dependencies
### ✍️ Summary
<div align='center'>

| SEVERITY                | DIRECT DEPENDENCIES                  | IMPACTED DEPENDENCY                  | FIXED VERSIONS                  | CVES                  |
| :---------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: |
| <br>  Medium | org.apache.poi:poi-scratchpad:4.1.1<br>pentaho:pentaho-platform-repository:10.2.0.0-SNAPSHOT<br>pentaho:pentaho-platform-extensions:10.2.0.0-SNAPSHOT | org.apache.poi:poi-scratchpad 4.1.1 | [5.2.1] | CVE-2022-26336 |

</div>

### 🔬 Research Details
**Description:**
A shortcoming in the HMEF package of poi-scratchpad (Apache POI) allows an attacker to cause an Out of Memory exception. This package is used to read TNEF files (Microsoft Outlook and Microsoft Exchange Server). If an application uses poi-scratchpad to parse TNEF files and the application allows untrusted users to supply them, then a carefully crafted file can cause an Out of Memory exception. This issue affects poi-scratchpad version 5.2.0 and prior versions. Users are recommended to upgrade to poi-scratchpad 5.2.1.
<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :x: Build failed in 13m 1s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5641/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[🚨 Frogbot scanned this pull request and found the below:](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>


## 📦 Vulnerable Dependencies
### ✍️ Summary
<div align='center'>

| SEVERITY                | DIRECT DEPENDENCIES                  | IMPACTED DEPENDENCY                  | FIXED VERSIONS                  | CVES                  |
| :---------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: |
| <br>  Medium | org.apache.poi:poi-scratchpad:4.1.1<br>pentaho:pentaho-platform-repository:10.2.0.0-SNAPSHOT<br>pentaho:pentaho-platform-extensions:10.2.0.0-SNAPSHOT | org.apache.poi:poi-scratchpad 4.1.1 | [5.2.1] | CVE-2022-26336 |

</div>

### 🔬 Research Details
**Description:**
A shortcoming in the HMEF package of poi-scratchpad (Apache POI) allows an attacker to cause an Out of Memory exception. This package is used to read TNEF files (Microsoft Outlook and Microsoft Exchange Server). If an application uses poi-scratchpad to parse TNEF files and the application allows untrusted users to supply them, then a carefully crafted file can cause an Out of Memory exception. This issue affects poi-scratchpad version 5.2.0 and prior versions. Users are recommended to upgrade to poi-scratchpad 5.2.1.
<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
<div align='center'>

[🚨 Frogbot scanned this pull request and found the below:](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>


## 📦 Vulnerable Dependencies
### ✍️ Summary
<div align='center'>

| SEVERITY                | DIRECT DEPENDENCIES                  | IMPACTED DEPENDENCY                  | FIXED VERSIONS                  | CVES                  |
| :---------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: |
| <br>  Medium | org.apache.poi:poi-scratchpad:4.1.1<br>pentaho:pentaho-platform-repository:10.2.0.0-SNAPSHOT<br>pentaho:pentaho-platform-extensions:10.2.0.0-SNAPSHOT | org.apache.poi:poi-scratchpad 4.1.1 | [5.2.1] | CVE-2022-26336 |

</div>

### 🔬 Research Details
**Description:**
A shortcoming in the HMEF package of poi-scratchpad (Apache POI) allows an attacker to cause an Out of Memory exception. This package is used to read TNEF files (Microsoft Outlook and Microsoft Exchange Server). If an application uses poi-scratchpad to parse TNEF files and the application allows untrusted users to supply them, then a carefully crafted file can cause an Out of Memory exception. This issue affects poi-scratchpad version 5.2.0 and prior versions. Users are recommended to upgrade to poi-scratchpad 5.2.1.
<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :x: Build failed in 7m 6s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5641/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]

**Review by rmansoor — APPROVED** [review]

**Pull request merged**